Gems/Atom/RPI/Code/Source/RPI.Builders/Material/MaterialTypeBuilder.cpp

```diff
@@ -327,7 +327,32 @@ namespace AZ
 
             if (auto settingsRegistry = AZ::SettingsRegistry::Get(); settingsRegistry != nullptr)
             {
-                settingsRegistry->GetObject(materialPipelines, "/O3DE/Atom/RPI/MaterialPipelineFiles");
+                const char* materialPipelineJsonPath = "/O3DE/Atom/RPI/MaterialPipelineFiles";
+
+                if (settingsRegistry->GetType(materialPipelineJsonPath) == SettingsRegistryInterface::Type::Object)
+                {
+                    // New method of adding material pipelines, allow for gems to specify their own material pipelines
+                    AZStd::map<AZStd::string, AZStd::vector<AZStd::string>> materialPipelinesMap;
+                    settingsRegistry->GetObject(materialPipelinesMap, materialPipelineJsonPath);
+                    for (const auto& [_ /*gemName*/, materialPipelinePaths] : materialPipelinesMap)
+                    {
+                        materialPipelines.insert(materialPipelines.end(), materialPipelinePaths.begin(), materialPipelinePaths.end());
+                    }
+                    AZStd::sort(materialPipelines.begin(), materialPipelines.end());
+                    materialPipelines.erase(AZStd::unique(materialPipelines.begin(), materialPipelines.end()), materialPipelines.end());
```

Comment:
If the serializer supports it, you could change the return type and the container to a set. Even though it was not sorted before, it should be sorted like you have it now to make sure that any of generated sub IDs stay in a deterministic order.

Because you're sorting and checking for uniqueness, you should probably also convert all of the paths to absolute paths before adding them to the container in case they are specified in different ways in different gems, or accidentally added more than once, so that we don't process the same material pipeline multiple times. I do not think that there are checks for that anywhere else and there should be.

You can use something like this to remove aliases or convert to absolute paths

        AZ::IO::FixedMaxPath pathWithoutAlias;
        AZ::IO::FileIOBase::GetInstance()->ReplaceAlias(pathWithoutAlias, AZ::IO::PathView{ path });
        return pathWithoutAlias.StringAsPosix();

Reply:
Changed return type to `AZStd::set<AZStd::string>` and use function `ResolvePath()` to resolve aliases and convert to absolute path.